### ISSUE TITLE
anvi-run-hmms does not check if the names inside genes.txt match with the names inside genes.hmm.gz

### ISSUE THREAD

**Issue by ozcan** [merenlab/anvio#815]
This is the actual problem behind https://github.com/merenlab/anvio/issues/814 reported by @ShaiberAlon.

I think anvi'o should directly read names from `genes.hmm.gz` instead of using names.txt. We can still keep the names.txt for informative purposes but do not use it in the codebase.

What do you think about this @meren?

**Comment by meren** [comment]
Thanks for finding this out, ozcan. It must have been such a headache. I agree we should scrutinize this a bit better.

But I think it would be better if we checked the `NAME` tags appear in `genes.hmm.gz` and make sure it perfectly matches to entries in `genes.txt` for external HMM collections that are used through the `-H` param :(

We can avoid that problem by focusing on `genes.hmm.gz`, but other programs later may utlize the `genes.txt` file since it is a standard file that can contain other kinds of information, being explicit and making sure that file matches to HMMs would be better in the long run.

My 2 cents.

**Comment by ozcan** [comment]
Ah I completely ignored the fact that `genes.txt` contains other information besides gene names. You are right, just checking if they matching should be enough. I will try to add some checks.

**Comment by jvhagey** [comment]
Hi @ozcan @ShaiberAlon 

I am currently using anvio5.1, previously I used anvio4 to build my contig.db and ran some custom HMMs with anvi-run-hmms in. I have migrated my contig.db to v12 with anvio5.1. I am running into the same problem reported here : 
https://groups.google.com/forum/#!searchin/anvio/info_dict%5Bhmm_search_source%5D%5Bhmm_item%5D$20$2B$3D$201%7Csort:date/anvio/phpyTO0ueGE/9XIcSlyMBAAJ

Basically, when running anvi-summarize I get the error: info_dict[hmm_search_source][hmm_item] += 1 
Looks like this issue might have been fixed with the release of anvio5.1, but if I ran anvi-run-hmms with anvio4 does that mean I need delete the hmms from the contig.db and rerun anvi-run-hmms with anvio5.1 to fix the bug? Or is there a less time consuming way to fix this? Thanks a million!
